sklearn/utils/_array_api.py

```diff
@@ -302,6 +302,12 @@ def __eq__(self, other):
     def isdtype(self, dtype, kind):
         return isdtype(dtype, kind, xp=self._namespace)
 
+    def maximum(self, x1, x2):
+        x1_np = _convert_to_numpy(x1, xp=self._namespace)
+        x2_np = _convert_to_numpy(x2, xp=self._namespace)
+        x_max = numpy.maximum(x1_np, x2_np)
+        return self._namespace.asarray(x_max, device=device(x1, x2))
```

Comment:
Which library(ies) need this?

Assuming that the 2023.12 version of the spec gets adopted by libraries in their future releases (see https://data-apis.org/array-api/2023.12/API_specification/generated/array_api.maximum.html), maybe we should only perform the numpy conversion-based fallback when `not hasattr(self._namespace, "maximum")` and call `self._namespace.maximum` when available.

Reply:
Note that `array-api-compat` should already provide `maximum` as per:

- https://github.com/data-apis/array-api-compat/pull/157

Maybe it's only in the dev branch and not yet released?

Reply:
This is only needed by `cupy.array_api` for which the _ArrayApiWrapper is used.

Reply:
And `array-api-compat` does not work in this case? It would be worth digging if this problem has already been reported or not on their tracker. If it is cross-link the issue number in an inline comment with a TODO to remove once the fix is released in `array-api-compat`. Otherwise create the issue and cross-link.